src/shims/tsd.h

```diff
@@ -228,6 +296,8 @@ _dispatch_thread_setspecific_packed_pair(pthread_key_t k1, pthread_key_t k2,
 #if DISPATCH_USE_DIRECT_TSD
 #define _dispatch_thread_port() ((mach_port_t)(uintptr_t)\
 		_dispatch_thread_getspecific(_PTHREAD_TSD_SLOT_MACH_THREAD_SELF))
+#elif DISPATCH_USE_THREAD_LOCAL_STORAGE
+#define _dispatch_thread_port() ((mach_port_t)(_dispatch_get_tsd_base()->tid))
```

Comment:
I really don't like this name, `_dispatch_thread_port()` is a mach concept, so we should ban the use of this on the linux port, and instead have `_dispatch_tid_self()` in lock.h expand to `(_dispatch_get_tsd_base()->tid)`.

and if there are calls to _dispatch_thread_port() in non mach code, we should most likely port them to `_dispatch_tid_self()` or guard the code with a `#if HAVE_MACH`

IOW that thing here should be in a `#if HAVE_MACH` this way:

```C
#if TARGET_OS_WIN32
#define _dispatch_thread_port() ((mach_port_t)0)
#elif HAVE_MACH
#if DISPATCH_USE_DIRECT_TSD
#define _dispatch_thread_port() ((mach_port_t)(uintptr_t)\
  		_dispatch_thread_getspecific(_PTHREAD_TSD_SLOT_MACH_THREAD_SELF))
#else
#define _dispatch_thread_port() pthread_mach_thread_np(_dispatch_thread_self())
#endif
#endif // HAVE_MACH
```